pkg/request/batchstore.go

```diff
@@ -28,14 +28,14 @@ type batch struct {
 	lock     sync.RWMutex
 	size     uint32
 	enqueued bool
-	m        sync.Map
```

Comment:
did we test whether this is faster or not? Can we test which of the two is faster? 

Additionally, I think this naive implementation should be improved. 
Let's have a constant set of maps (such as the number of cores we have) and when we store, we first hash the key into one of the maps and then lock only that map and insert to it. 

Then let's compare which is faster among the three.

Reply:
Did not test if it is faster yet.
We can try what you are suggesting.
I want to first integrate the pool as is and then we could try other things.